src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonPropertyInfoOfTConverter.cs

```diff
@@ -0,0 +1,190 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+using System.Reflection;
+using System.Text.Json.Serialization;
+
+namespace System.Text.Json
+{
+    /// <summary>
+    /// Represents a strongly-typed property to prevent boxing and to create a direct delegate to the getter\setter.
+    /// </summary>
+    internal sealed class JsonPropertyInfo<TConverter> : JsonPropertyInfo
+    {
+        public Func<object, TConverter>? Get { get; private set; }
+        public Action<object, TConverter>? Set { get; private set; }
+
+        public JsonConverter<TConverter> Converter { get; internal set; } = null!;
```

Comment:
Yes most of the logic is from that file. The converter now is never null here after creation since everything is a converter now (objects and collections).

Reply:
> The converter now is never null here after creation since

Nice. I like that :)